lightning/src/events/mod.rs

```diff
@@ -1544,16 +1550,20 @@ impl Writeable for Event {
 					(0, payment_hash, required),
 					(1, receiver_node_id, option),
 					(2, payment_secret, option),
-					(3, via_channel_id, option),
+					// legacy field
+					// (3, via_channel_id, option),
 					(4, amount_msat, required),
-					(5, via_user_channel_id, option),
+					// legacy field
+					// (5, via_user_channel_id, option),
```

Comment:
I think we should still write these fields for the sake of users who downgrade (can always write the first item in the vec). Also would be nice to comment in what version the fields became legacy.